src/traces/scatter/defaults.js

```diff
@@ -63,6 +66,11 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         if(!subTypes.hasLines(traceOut)) handleLineShapeDefaults(traceIn, traceOut, coerce);
     }
 
+    if(traceOut.fill === 'tonext' || traceOut.fill === 'toself') {
+        dfltHoverOn.push('fills');
```

Comment:
I'm having second thoughts about making `hoveron: 'points+fills'` the default for `scatter` (my mistake).

With `hovermode: 'x'` by default the `'fills'` in `hoveron: 'points+fills'` is hardly noticeable for most polygons. 

Moreover, overriding the `layout.hovermode` default based on `fill` would a little too intrusive I think. 

While `hovermode: 'closest'` by default for `scatterternary`, I'm not sure it warrants a different default than its `scatter` cousin. 

So, I'm leaning towards making `hoveron: 'points'` the default for `scatter` and `scatterternary` regardless of the `fill` value.

Thoughts?